### PR TITLE
Logistic regression 

### PR DESCRIPTION
**Pull request by pascoesean**
Adding Logistic Regression/component weight analysis for Lupus dataset
@andrewram4287

### CODE REVIEW

File: sccp/figures/figureS2.py [hunk 1]
```diff
@@ -55,20 +34,20 @@ def makeFigure():
     cmp = 11
 
     _, factors, projs, = openPf2(rank = 39, dataName = 'lupus')
```
Comment (pascoesean) [inline sccp/figures/figureS2.py:36]: this is where build is failing in #167 because projs file isn't uploaded (too big for rank 39)

**Comment by pascoesean** [comment]
@andrewram4287 if you can check this out now it should be good as a version of my logistic regression stuff-- I can add some comments to the code itself on github since its a larger PR

File: sccp/figures/common.py [hunk 2]
```diff
@@ -99,7 +103,7 @@ def genFigure():
     print(f"Figure {sys.argv[1]} is done after {time.time() - start} seconds.\n")
 
 
-def plotFactors(factors, data: Pf2X, axs, reorder=tuple(), trim=tuple(), saveGenes=False, row_colors= None):
+def plotFactors(factors, data: Pf2X, axs, reorder=tuple(), trim=tuple(), saveGenes=False, cond_group_labels= None):
```
Comment (pascoesean) [inline sccp/figures/common.py:106]: more generalizable

File: sccp/figures/common.py [hunk 3]
```diff
@@ -17,6 +17,10 @@
 import os
 from os.path import join
 from pandas.plotting import parallel_coordinates as pc
+from parafac2 import parafac2_nd
+from sklearn.linear_model import LogisticRegressionCV, LogisticRegression
+from sklearn.metrics import roc_auc_score, RocCurveDisplay
```
Comment (pascoesean) [inline sccp/figures/common.py:22]: need to be able to run Pf2 for different ranks in `testPf2Ranks`; need to be able to do cross validated regression/fit after, and calculate ROC AUC

File: sccp/figures/common.py [hunk 4]
```diff
@@ -463,19 +471,20 @@ def plotUMAP_ct(labels, pf2Points, ax):
         xlabel="UMAP1",
         title="Pf2-Based Decomposition: Label Cell Types")
     
-def plotCellStateViolins(projections, cell_types, cell_state: int, ax):
-    all_cell_projs = pd.DataFrame(projections)
-    cell_state_n = pd.concat([all_cell_projs.iloc[:, (cell_state - 1)], cell_types], axis = 1)
-    cell_state_n.columns.values[0] = "contribution"
+def plotCompViolins(projection_B, cell_types, component: int, ax):
```
Comment (pascoesean) [inline sccp/figures/common.py:474]: replace proj matrix with proj @ B

File: sccp/figures/figureS1.py [hunk 5]
```diff
@@ -10,34 +10,36 @@
...
+    savePf2(weights, factors, projs, dataName = "lupus")
+
     plotFactors(
-        factors, lupus_tensor, ax[0:3], reorder=(0, 2), trim=(2,), row_colors=row_colors
+        factors, lupus_tensor, ax[0:3], reorder = (0,2), trim=(2,), cond_group_labels = group_labs
     )
 
     plotWeight(weights, ax[3])
-    ax[3].set_title("Weight of Each Componenet")
+    ax[3].set_title("Weight of Each Component")
```
Comment (pascoesean) [inline sccp/figures/figureS1.py:43]: finally fix this <3

File: sccp/figures/figureS2.py [hunk 6]
```diff
@@ -30,29 +27,22 @@ def makeFigure():
...
-    projs = np.concatenate(projs, axis=0)
+    _, cell_types, _ = load_lupus_data()  # don't need to get patient color mappings
+    rank = 39
+    cellState = 11
+    cmp = 11
+
+    _, factors, projs, = openPf2(rank = rank, dataName = 'lupus', optProjs=True)
+
+
+    proj_B = projs @ factors[1]
```
Comment (pascoesean) [inline sccp/figures/figureS2.py:38]: replace running pf2 with openPf2 (note that savePf2 concatenates the projections; so we don't need that step again here)

File: sccp/figures/figureS3b.py [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+    _, _, group_labs = load_lupus_data() # don't need to grab cell types here
+    
+
+    _, factors, _, = openPf2(rank = rank, dataName = 'lupus')
+        
+    A_matrix = factors[0]
+        
+    plotCmpRegContributions(A_matrix, group_labs.to_numpy(), 50, rank, ax[0])
+
```
Comment (pascoesean) [inline sccp/figures/figureS3b.py:40]: Make this figure:
figureS3b_comp39

File: sccp/figures/figureS3.py [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+    
+
+    ranks_to_test = [5, 10]
+    penalties_to_test = [10, 50, 200, 1000]
+
+    results = testPf2Ranks(lupus_tensor, group_labs, ranks_to_test, 
+                           penalties_to_test=penalties_to_test)
+    
+    plotPf2RankTest(results, ax[0])
+
```
Comment (pascoesean) [inline sccp/figures/figureS3.py:43]: Make this figure (or some version of it)

figureS3

File: sccp/imports/scRNA.py [hunk 9]
```diff
@@ -111,7 +111,7 @@ def import_pancreas_all(tensor=True, method=str()):
     return pancreas, methods
 
 
-def load_lupus_data(third_axis="ind_cov", every_n=4):
+def load_lupus_data(third_axis="ind_cov", every_n=1, give_batch = False):
```
Comment (pascoesean) [inline sccp/imports/scRNA.py:114]: add ability to return what processing cohort each person was in; this is used here but I might update it in a new version of this function i'm working on in my new branch that allows you to specify what you want the 'cell_type' column to be

File: sccp/figures/common.py [hunk 10]
```diff
@@ -142,13 +146,16 @@ def plotFactors(factors, data: Pf2X, axs, reorder=tuple(), trim=tuple(), saveGen
...
 
-        if i == 0 and not (row_colors is None):
+        if i == 0 and not (cond_group_labels is None):
             # add little boxes to denote SLE/healthy rows
             axs[i].tick_params(axis='y', which='major', pad=20, length=0) # extra padding to leave room for the row colors
-            for iii, color in enumerate(row_colors):
-                axs[i].add_patch(plt.Rectangle(xy=(-0.05, iii), width=0.05, height=1, color=color, lw=0,
-                             transform=axs[i].get_yaxis_transform(), clip_on=False))
-
+            for iii, status in enumerate(cond_group_labels):
```
[the remark this answers is not in the document]
Comment (pascoesean) [reply sccp/figures/common.py:152]: yeah I think initially it required an input color list and now it just takes a list of labels; but you're right I think I can make this more intuitive by kinda combining both
